lib/java/src/org/apache/thrift/transport/TSaslTransport.java

```diff
@@ -310,14 +310,13 @@ public void open() throws TTransportException {
         underlyingTransport.close();
       }
     } catch (TTransportException e) {
-      /*
-       * If there is no-data or no-sasl header in the stream, throw a different
-       * type of exception so we can handle this scenario differently.
-       */
+      // If there is no-data or no-sasl header in the stream,
+      // log the failure, and clean up the underlying transport.
       if (!readSaslHeader && e.getType() == TTransportException.END_OF_FILE) {
         underlyingTransport.close();
-        LOGGER.debug("No data or no sasl data in the stream");
-        throw new TSaslTransportException("No data or no sasl data in the stream during negotiation", e);
+        if (LOGGER.isDebugEnabled()) {
```

Comment:
Is "isDebugEnabled" necessary when LOGGER.debug is used?

Reply:
You're right. This check is not necessary for simple logging. 

If a heavier object (i.e. harder to convert `toString()`), it might make sense to do a pre-check. In this case, since it's a simple string, I should forgo the `isDebugEnabled()` check.